apps/condo/domains/news/access/NewsItemScope.js

```diff
@@ -37,33 +36,65 @@ async function canReadNewsItemScopes (attrs) {
     }
 }
 
-async function canManageNewsItemScopes ({ authentication: { item: user }, originalInput, operation, itemId }) {
+async function canManageNewsItemScopes ({ authentication: { item: user }, originalInput, operation, itemId, itemIds }) {
     if (!user) return throwAuthenticationError()
     if (user.deletedAt) return false
+
+    const isBulkRequest = Array.isArray(originalInput)
+    const isSoftDeleteOperation = operation === 'update'
+        && (
+            isBulkRequest
+                ? (Array.isArray(itemIds) && originalInput.every(item => isSoftDelete(get(item, 'data'))))
+                : (itemId && isSoftDelete(originalInput))
+        )
+
+    if (operation !== 'create' && !isSoftDeleteOperation) return false
     if (user.isAdmin) return true
 
     if (user.type === STAFF) {
-        let newsItemId
+        let organizationIds
 
         if (operation === 'create') {
-            newsItemId = get(originalInput, ['newsItem', 'connect', 'id'])
-        }
-
-        if (operation === 'update') {
-            // allow soft-deleting only
-            if (!isEmpty(omit(originalInput, ['deletedAt', 'dv', 'sender']))) {
-                return false
+            if (isBulkRequest) {
+                const newsItemIds = uniq(originalInput.map(item => get(item, 'data.newsItem.connect.id')))
+                const newsItems = await find('NewsItem', {
+                    id_in: newsItemIds,
+                    deletedAt: null,
+                })
+                if (newsItemIds.length !== newsItems.length) return false
+                organizationIds = uniq(newsItems.map(item => get(item, 'organization')))
+            } else {
+                const newsItemId = get(originalInput, 'newsItem.connect.id')
+                const newsItem = await getById('NewsItem', newsItemId)
+                if (!newsItem) return false
+                organizationIds = [get(newsItem, 'organization')]
             }
-            const newsItemScope = await getById('NewsItemScope', itemId)
-            newsItemId = newsItemScope.newsItem
-        }
+        } else if (isSoftDeleteOperation) {
+            if (isBulkRequest) {
+                if (!itemIds || !Array.isArray(itemIds)) return false
+                if (itemIds.length !== uniq(itemIds).length) return false
```

Comment:
Why is this condition needed?

Reply:
It will be strange if you delete the same object many times.
We can skip this check, but this means that we will do some extra operations when we can avoid them